src/Control/Monad/Trans/Fail.hs

```diff
@@ -56,6 +56,7 @@ import Control.Exception
 import Control.Monad.Catch (MonadThrow (throwM))
 import Control.Monad.Cont
 import Control.Monad.Except
+import Control.Monad.Fix
```

Comment:
I think this will be the easiest way to avoid all of the warnings for pre mtl-2.3 versions.
```suggestion
#if MIN_VERSION_mtl(2,3,0)
import Control.Monad.Fix
#endif
```

An alternative would be to make all of the mtl related imports explicit. I'll let you choose the solution.